src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -1304,6 +1307,31 @@ module Base = struct
                  let failed = Boolean.(failed1 ||| (is_new &&& failed2)) in
                  `Flagged (res, failed)))
       in
+      let `Min_balance _, timing =
+        !([%with_label "Check snapp timing"]
+            (let open Tick in
+            let balance_check ok =
+              add_check ~label:__LOC__ !(Boolean.any [ ok; is_receiver ]) ;
+              return ()
+            in
+            let timed_balance_check ok =
+              add_check ~label:__LOC__ !(Boolean.any [ ok; is_receiver ]) ;
+              return ()
+            in
+            (* NB: We perform the check here with the final balance and a zero
+               amount. This allows this to serve dual purposes:
+               * if the balance has decreased, this checks that it isn't below
+                 the minimum;
+               * if the account is new and this party has set its timing info,
+                 this checks that the timing is valid.
+               The balance and txn_amount are used only to find the resulting
+               balance, so using the result directly is equivalent.
+            *)
+            check_timing ~balance_check ~timed_balance_check
```

Comment:
We should not check if timing is valid if the account is new, right? To make it equivalent to what is in transaction logic. If an account has been created with balance less than min balance then the snark would fail

Reply:
I've updated the transaction logic to check that the timing is valid too. We make the assumption that the timing info is valid in several places -- for example, when receiving fee transfers in the old circuit -- so allowing an account with invalid timing info into the ledger could result in an unrecoverable network stall.

Reply:
I was looking at the genesis ledger flow and we don't check there. There is a PR that explicitly removes that check https://github.com/MinaProtocol/mina/pull/6722

Reply:
Perhaps we should revert that.